File: problems/C++/34. Find First and Last Position of Element in Sorted Array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int l = find(nums, target, true);
        int r = find(nums, target, false);
        return {l, r};
    }

private:
    int find(const vector<int>& nums, int target, bool first) {
        int l = 0, r = nums.size() - 1;
        int idx = -1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (nums[mid] > target) {
                r = mid - 1;
            } else if (nums[mid] < target) {
                l = mid + 1;
            } else {
                idx = mid;
                if (first) {
                    r = mid - 1;
                } else {
                    l = mid + 1;
                }
            }
        }
        return idx;
    }
};

File: problems/Python/34. Find First and Last Position of Element in Sorted Array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l = self.find(nums, target, True)
        r = self.find(nums, target, False)
        return [l, r]

    def find(self, nums: List[int], target: int, first: bool) -> int:
        l, r = 0, len(nums) - 1
        idx = -1
        while l <= r:
            mid = (l + r) // 2
            if nums[mid] > target:
                r = mid - 1
            elif nums[mid] < target:
                l = mid + 1
            else:
                idx = mid
                if first:
                    r = mid - 1
                else:
                    l = mid + 1
        return idx